packages/framework/src/Model/Product/Search/ProductElasticsearchRepository.php

```diff
@@ -47,27 +47,35 @@ class ProductElasticsearchRepository
      * @param \Elasticsearch\Client $client
      * @param \Shopsys\FrameworkBundle\Model\Product\Search\ProductElasticsearchConverter $productElasticsearchConverter
      * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\ElasticsearchStructureManager $elasticsearchStructureManager
+     * @param \Shopsys\FrameworkBundle\Model\Product\Search\FilterQueryFactory|null $filterQueryFactory
      */
     public function __construct(
         string $indexPrefix,
         Client $client,
         ProductElasticsearchConverter $productElasticsearchConverter,
-        ElasticsearchStructureManager $elasticsearchStructureManager
+        ElasticsearchStructureManager $elasticsearchStructureManager,
+        ?FilterQueryFactory $filterQueryFactory = null
     ) {
         $this->indexPrefix = $indexPrefix;
         $this->client = $client;
         $this->productElasticsearchConverter = $productElasticsearchConverter;
         $this->elasticsearchStructureManager = $elasticsearchStructureManager;
-        $this->filterQueryFactory = $this->createFilterQueryFactory();
+        $this->filterQueryFactory = $filterQueryFactory;
```

Comment:
```suggestion
        $this->filterQueryFactory = $filterQueryFactory ?? $this->createFilterQueryFactory();
```
It seems that you have misunderstood the usage - `createFilterQueryFactory` is not a setter injector.

Reply:
Done